src/main/java/seedu/address/model/roles/Role.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.roles;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Role in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidRoleName(String)}
+ */
+public class Role {
+
+    public static final String MESSAGE_CONSTRAINTS = "Role names should be alphanumeric with whitespace";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}\\s]+";
+
+    public final String roleName;
+
+    /**
+     * Constructs a {@code Role}.
+     *
+     * @param roleName A valid role name.
+     */
+    public Role(String roleName) {
+        requireNonNull(roleName);
+        checkArgument(isValidRoleName(roleName), MESSAGE_CONSTRAINTS);
+        this.roleName = roleName;
+    }
+
+    /**
+     * Returns true if a given string is a valid role name.
+     */
+    public static boolean isValidRoleName(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof Role)) {
+            return false;
+        }
+
+        Role otherRole = (Role) other;
+        return roleName.equals(otherRole.roleName);
+    }
+
+    @Override
+    public int hashCode() {
+        return roleName.hashCode();
+    }
+
+    /**
+     * Format state as text for viewing.
+     */
+    public String toString() {
+        return '[' + roleName + ']';
+    }
+
+}
```

Comment:
Compared with the original tag file, no issue here. But better for someone else look at it again

Reply:
Okay, do you suggest we wait for somebody else to look through these changes, or do I merge it now?

Reply:
Better for another guy look through it again